src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs

```diff
@@ -80,7 +78,19 @@ public void Dispose()
         /// </summary>
         internal void Flush(TimeSpan timeout)
         {
-            this.DequeueAndSend(timeout);
+#if !CORE_PCL
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+#endif
+                this.DequeueAndSend(timeout);
```

Comment:
I want to do it once around the "while" instead of doing it at every cycle of the loop